File: 0747-largest-number-at-least-twice-of-others/0747-largest-number-at-least-twice-of-others.cpp
class Solution {
public:
    int dominantIndex(vector<int>& nums) {
        map<int,int> mp;
        for(int i = 0; i < nums.size(); i++){
            mp.insert({nums[i]*-1, i});
        }
        map<int,int>::iterator it = mp.begin();
        int num1,num2, num1Index;
        num1 = -1 * it->first, num1Index = it->second;
        it++;
        num2 = -1 * it->first;
        if(num1 >= num2 * 2){
            return num1Index;
        }
        return -1;
    }
};

File: 0747-largest-number-at-least-twice-of-others/0747-largest-number-at-least-twice-of-others.cs
public class Solution {
    public int DominantIndex(int[] nums) {
        //  https://leetcode.com/problems/largest-number-at-least-twice-of-others/solutions/2865024/c-simple-solution-beats-97-24
        int max = nums.Max();
        for(int i = 0; i < nums.Length; i++){
            if(nums[i] != max && nums[i] * 2 > max) return -1;
        }
        return Array.IndexOf(nums, max);
    }
}